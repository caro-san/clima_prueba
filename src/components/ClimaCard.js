import React from 'react';
import {Card} from 'react-bootstrap';
import img from '../img/9846.jpg';
var moment = require('moment');

const ClimaCard = ({dt, temp_min, temp_max, temp_main, main, icon}) => {
  // create a date object with Date class constructor
  const newDate = new Date();
    newDate.setTime(dt)

  return (
    <>
    <Card className="mb-2" style={{backgroundImage: "url("+img+")"}}>
      <Card.Body>
        <p style={{fontSize:'3rem'}}>{temp_main}</p>
        <Card.Subtitle className="mb-1 text-muted">{moment(newDate).format('dddd')}</Card.Subtitle>
      </Card.Body>
      <Card.Img
        variant="top"
        src={`http://openweathermap.org/img/wn/${icon}@2x.png`}
      />
      <Card.Body>
        <Card.Title>{main}</Card.Title>
        <p>
          {newDate.toLocaleDateString()} - {newDate.toLocaleTimeString()}
        </p>
        <p>Min: {temp_min}</p>
        <p>Max: {temp_max}</p>
      </Card.Body>
    </Card>
    </>
  );
};
export default ClimaCard;