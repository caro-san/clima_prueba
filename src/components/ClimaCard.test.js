import React from 'react';
import {render} from '@testing-library/react';
import ClimaCard from "./ClimaCard";

describe("ClimaCard", () => {
    it("should render", () => {
    const props = {
        dt_txt: "2020-12-06 18:00:00",
        temp_min: 20,
        temp_max: 22,
        temp_main: 21,
        main: "Clear",
        icon: "01d",
    };
    const { container } = render(<ClimaCard {...props} />);
    expect(container).toMatchSnapshot();
    });
});