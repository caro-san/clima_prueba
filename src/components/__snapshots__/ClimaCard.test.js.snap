// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ClimaCard should render 1`] = `
<div>
  <div
    class="mb-2 card"
    style="background-image: url(9846.jpg);"
  >
    <div
      class="card-body"
    >
      <p
        style="font-size: 3rem;"
      >
        21
        ºC
      </p>
      <div
        class="mb-1 text-muted card-subtitle h6"
      >
        Invalid date
      </div>
    </div>
    <img
      class="card-img-top"
      src="http://openweathermap.org/img/wn/01d@2x.png"
    />
    <div
      class="card-body"
    >
      <div
        class="card-title h5"
      >
        Clear
      </div>
      <p>
        Invalid date
      </p>
      <p>
        Min: 
        20
        ºC
      </p>
      <p>
        Max: 
        22
        ºC
      </p>
    </div>
  </div>
</div>
`;
